pkg/bpf/bpffs_linux.go

```diff
@@ -181,7 +181,7 @@ func checkOrMountCustomLocation(bpfRoot string, printWarning bool) error {
 	// If the custom location has no mount, let's mount BPFFS there.
 	if !mounted {
 		setMapRoot(bpfRoot)
-		if err := mountFS(printWarning); err != nil {
+		if err := mountFS(true); err != nil {
```

Comment:
How does mounting at a custom location work if we don't enable bidirectional mount at the custom mount point?

Reply:
If I understood the question, it would work the same way it before these changes, it's expected users to mount the bpf fs in the host and pass the right path into Cilium DaemonSet.

Reply:
>it's expected users to mount the bpf fs in the host and pass the right path into Cilium DaemonSet.

How does that work? `/sys/fs/bpf path` seems to be hard-coded.

Reply:
Hmm yeah it doesn't look like there's a helm flag for it so users would need to change it manually.

Reply:
If you remove `checkOrMountFS`, you can get rid of the `printWarning` related changes you had to do.

Reply:
See https://github.com/cilium/cilium/pull/16656#issuecomment-872665088